spec/features/users/sign_in_spec.rb

```diff
@@ -431,6 +431,38 @@
         end
       end
     end
+
+    context 'with sp' do
+      scenario 'redirects to home page with error' do
```

Comment:
Nit: I might include the detail about maintaining the SP metadata somewhere in this description, since that's primarily what we're interested in with this test.

```suggestion
      scenario 'redirects to home page with error and preserves sp metadata' do
```

or:

```suggestion
      scenario 'redirects to home page with error and preserves branded experience' do
```